ot/lp/emd_wrap.pyx

```diff
@@ -12,81 +12,33 @@ cimport numpy as np
 
 cimport cython
 
+import warnings
 
 
 cdef extern from "EMD.h":
-    int EMD_wrap(int n1,int n2, double *X, double *Y,double *D, double *G, double *cost, int max_iter)
-    cdef enum ProblemType: INFEASIBLE, OPTIMAL, UNBOUNDED
+    int EMD_wrap(int n1,int n2, double *X, double *Y,double *D, double *G, double* alpha, double* beta, double *cost, int maxIter)
+    cdef enum ProblemType: INFEASIBLE, OPTIMAL, UNBOUNDED, MAX_ITER_REACHED
 
 
+def check_result(result_code):
+    if result_code == OPTIMAL:
+        return None
 
-@cython.boundscheck(False)
-@cython.wraparound(False)
-def emd_c( np.ndarray[double, ndim=1, mode="c"] a,np.ndarray[double, ndim=1, mode="c"]  b,np.ndarray[double, ndim=2, mode="c"]  M, int max_iter):
-    """
-        Solves the Earth Movers distance problem and returns the optimal transport matrix
-
-        gamm=emd(a,b,M)
-
-    .. math::
-        \gamma = arg\min_\gamma <\gamma,M>_F
-
-        s.t. \gamma 1 = a
-
-             \gamma^T 1= b
-
-             \gamma\geq 0
-    where :
+    if result_code == INFEASIBLE:
+        message = "Problem infeasible. Check that a and b are in the simplex"
+    elif result_code == UNBOUNDED:
+        message = "Problem unbounded"
+    elif result_code == MAX_ITER_REACHED:
+        message = "numItermax reached before optimality. Try to increase numItermax."
```

Comment:
numItermax

git grep numItermax

is your friend.

Reply:
Thank you, I'm rather new to git, I didn't know this one. My ide is supposed to rename variable names in strings but I guess it doesn't work well with cython...

It seems that there are a lot of variables called numIterMax across the repository, though I don't think the renaming of all of them should be done in this PR.

Reply:
Hello

Yes we have some renaming to do but it is not the object of this pr.

I'm ok with cleaning names in emd-c but the EMD and emd2 function should stick to the old parameter names because even if they do not follow convention they are consistent across the toolbox.

We will open an issue/or to discuss renaming and proper deprecation procedure for parameters. I would like to merge the current  price and release 0.3 shortly.

Reply:
I reverted max_iter to its old name for now.